src/components/Form.js

```diff
@@ -61,8 +61,7 @@ export default class Form extends Component {
 
   onChange = (formData, options={validate: false}) => {
     const liveValidate = this.props.liveValidate || options.validate;
-    const errors = liveValidate ? this.validate(formData) :
-                                  this.state.errors;
+    const errors = liveValidate ? this.validate(formData) : [];
```

Comment:
Added live validation toggler to the playground, you're definitely right wrt surprising behavior. Il revamp the patch.